tests/test_main.py

```diff
@@ -91,6 +91,16 @@ def test_default_language(self):
         assert handler.env.lang == 'fr'  # first language in the options list
         assert django.utils.translation.get_language() == 'fr'
 
+    def test_content_security_policy_for_react(self):
+        """Verify CSP is set when the React UI is enabled."""
+        config.set(enable_react_ui=True)
+        request = setup_request('/')
+        response = webapp.Response()
+        handler = main.Main(request, response)
+        handler.get()
+        assert 'Content-Security-Policy' in response.headers
+        assert 'nonce-' in response.headers['Content-Security-Policy']
```

Comment:
It might be worthwhile to assert the full nonce in the header matches the response body, since that's what the browser will work against. Perhaps also verify the nonce is not just `"nonce-"`.

Reply:
Thanks, done. Since mocking in Python is (usually) ridiculously easy I mocked out generate_random_key and have it verifying the specific value.